license_solver/classifiers.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+# solver-license-job
+# Copyright(C) 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Class download classifiers and extract data from them."""
+
+import re
+import sys
+import urllib.request
+
+
+def _get_abbreviation(classifier: str) -> list:
+    """Abbreviation for license name."""
+    abbreviation = list()
+    start = classifier.find("(")
+    end = classifier.find(")")
+    if start != -1 and end != -1:
+        txt = classifier[classifier.find("(") + 1 : classifier.find(")")]
+        abbreviation.append(txt)
+    return abbreviation
+
+
+def _get_name_license(classifier: str) -> str:
+    """Get licence name from classifier string."""
+    data = classifier.split("::")
+
+    if len(data) > 0 and len(data[len(data) - 1]) > 1:
+        return data[len(data) - 1].strip()[0:]
+    else:
+        return ""
+
+
+class Classifiers:
+    """Class detect all classifiers from downloaded data."""
+
+    _received_text: str = ""
+    classifiers: list = list()
+    classifiers_list: list = list()
+
+    def __attrs_post_init__(self):
+        """INIT method."""
+        self._download_classifiers()
+        self._cmp_sets_of_data()
+        self._extract_classifiers()
+
+    def _download_classifiers(self) -> None:
+        """Download classifiers from PyPI."""
+        url = "https://pypi.org/pypi?%3Aaction=list_classifiers"
+        try:
+            response = urllib.request.urlopen(url)
```

Comment:
You can use [requests](https://docs.python-requests.org/en/master/), it has more human friendly API:

```python
response.requests.get(url)
response.raise_for_status()
```